### ISSUE TITLE
No true positives after training 

### ISSUE THREAD

**Issue by HariP1234** [thtrieu/darkflow#645]
Hi,
I am new to object detection. I am trying to train darkflow model to detect license plates in real time. But no matter how much I train it, I couldn't get correct results. There are a lot of bounding boxes but none of them are on correctly enclosing the object. Here is what I have done so far:
I used a dataset of size 500. Created annotation xml files in PascalVOC format for each of them using LabelImg. One of the images and its annotated form are given below.
![image1](https://user-images.githubusercontent.com/37569736/37667350-ca44c7d6-2c87-11e8-85e7-4bbe4de6ccbe.jpg)
![annotated_image1](https://user-images.githubusercontent.com/37569736/37667447-07c09c52-2c88-11e8-90d9-feaec6159bef.jpg)
I am using pre-trained tiny-yolo-voc.weights. I copied the tiny-yolo-voc.cfg to tiny-yolo-voc-1c.cfg. I changed no:of classes in my cfg file to 1 and filters to 30. Then I followed the steps:(I am using Anaconda prompt)
1. cd "PATH"\darkflow-master
2. python setup.py build_ext --inplace
3. python flow --model cfg/tiny-yolo-voc-1c.cfg --load bin/tiny-yolo-voc.weights --train --annotation train/Annotations --dataset train/Images --gpu 1.0
I tried training in different ways: (In all cases epoch=200, batch size=16, learning rate=5, height and width were adjusted according to image each time)
1. First I resized the images to 288x288 trained using 100 imgaes. Moving ave loss was around 0.06.
2. I then trained 500 images of 640x480 size for 200 epochs. Took 6200 steps and 23 hours. Again loss was fluctuating between 0.08 and 0.04.
3. I then started training repeatedly by increasing the no:of images by 10 each time. Size was 288x288. Again loss was less than 0.1.
But in all the cases when I input an image(one from the trained images) for testing I get the following result for thresholds 0.5 and 0.6 
python flow --imgdir sample_img/ --model cfg/tiny-yolo-voc-1c.cfg --threshold 0.5 --load bin/tiny-yolo-voc.weights --gpu 1.0
![sample_license_plate](https://user-images.githubusercontent.com/37569736/37668667-0a39a0e8-2c8b-11e8-8fe2-98c2279f4e6b.jpg)
![sample_license_plate](https://user-images.githubusercontent.com/37569736/37668732-2b10d750-2c8b-11e8-912d-b2601d795944.jpg)
What am I doing wrong? How can I get correct detection? I need to implement this in real time and I can't even get correct results with static images. Please help me.

**Comment by ssusie** [comment]
I am not sure at all, but having only one class could mess up lot of things. Try with three classes first and see what happens. Please also comment here so we know if the issue was in number of classes or in something different. I am also trying to make it work with only two classes, but haven't succeeded yet...

**Comment by zsfarkas** [comment]
+1

We have the same effect with 3 classes...

**Comment by Pele324** [comment]
Are these many bounding boxes similar over different images? If yes then I had this problem and it is caused by incorrect settings in the .cfg file. I solved the problem by changing the filter size in the output layer to filter = num * (classes + 5) like https://github.com/thtrieu/darkflow#training-on-your-own-dataset
